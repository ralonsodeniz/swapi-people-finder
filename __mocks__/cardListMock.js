export default [
  {
    name: 'Luke Skywalker',
    imageUrl: 'http://facetheforce.today/luke',
    gender: 'male',
    birthYear: '3523',
    eyeColor: 'blue',
  },
  {
    name: 'C-3PO',
    imageUrl: 'http://facetheforce.today/c3po',
    gender: 'male',
    birthYear: '3523',
    eyeColor: 'blue',
  },
  {
    name: 'R2-D2',
    imageUrl: 'http://facetheforce.today/r2d2',
    gender: 'male',
    birthYear: '3523',
    eyeColor: 'blue',
  },
  {
    name: 'Darth Vader',
    imageUrl: 'http://facetheforce.today/darthvader',
    gender: 'male',
    birthYear: '3523',
    eyeColor: 'blue',
  },
  {
    name: 'Leia Organa',
    imageUrl: 'http://facetheforce.today/leia',
    gender: 'male',
    birthYear: '3523',
    eyeColor: 'blue',
  },
  {
    name: 'Obi-Wan Kenobi',
    imageUrl: 'http://facetheforce.today/obiwan',
    gender: 'male',
    birthYear: '3523',
    eyeColor: 'blue',
  },
  {
    name: 'Anakin Skywalker',
    imageUrl: 'http://facetheforce.today/anakin',
    gender: 'male',
    birthYear: '3523',
    eyeColor: 'blue',
  },
];
