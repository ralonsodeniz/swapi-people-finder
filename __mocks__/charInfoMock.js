export const c3p0 = {
  name: 'C-3PO',
  height: '167',
  mass: '75',
  hairColor: 'n/a',
  skinColor: 'gold',
  eyeColor: 'yellow',
  birthYear: '112BBY',
  gender: 'n/a',
};

export const darthVader = {
  name: 'Darth Vader',
  height: '202',
  mass: '136',
  gender: 'male',
  hairColor: 'none',
  skinColor: 'white',
  eyeColor: 'yellow',
  birthYear: '41.9BBY',
};
