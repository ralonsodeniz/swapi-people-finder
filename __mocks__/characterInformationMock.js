export default {
  name: 'C-3PO',
  height: '55',
  mass: '35',
  hairColor: 'blond',
  skinColor: 'fair',
  eyeColor: 'blue',
  birthYear: '3523',
  gender: 'male',
};
