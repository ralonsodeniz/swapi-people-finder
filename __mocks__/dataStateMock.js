export default {
  loadingData: false,
  characterCount: 87,
  nextEndpoint: 'https://swapi.co/api/people/?page=2&search=',
  previousEndpoint: null,
  searchArray: [
    {
      name: 'Luke Skywalker',
      height: '172',
      mass: '77',
      gender: 'male',
      hairColor: 'blond',
      skinColor: 'fair',
      eyeColor: 'blue',
      birthYear: '19BBY',
    },
    {
      name: 'C-3PO',
      height: '167',
      mass: '75',
      gender: 'n/a',
      hairColor: 'n/a',
      skinColor: 'gold',
      eyeColor: 'yellow',
      birthYear: '112BBY',
    },
    {
      name: 'R2-D2',
      height: '96',
      mass: '32',
      gender: 'n/a',
      hairColor: 'n/a',
      skinColor: 'white, blue',
      eyeColor: 'red',
      birthYear: '33BBY',
    },
    {
      name: 'Darth Vader',
      height: '202',
      mass: '136',
      gender: 'male',
      hairColor: 'none',
      skinColor: 'white',
      eyeColor: 'yellow',
      birthYear: '41.9BBY',
    },
    {
      name: 'Leia Organa',
      height: '150',
      mass: '49',
      gender: 'female',
      hairColor: 'brown',
      skinColor: 'light',
      eyeColor: 'brown',
      birthYear: '19BBY',
    },
    {
      name: 'Owen Lars',
      height: '178',
      mass: '120',
      gender: 'male',
      hairColor: 'brown, grey',
      skinColor: 'light',
      eyeColor: 'blue',
      birthYear: '52BBY',
    },
    {
      name: 'Beru Whitesun lars',
      height: '165',
      mass: '75',
      gender: 'female',
      hairColor: 'brown',
      skinColor: 'light',
      eyeColor: 'blue',
      birthYear: '47BBY',
    },
    {
      name: 'R5-D4',
      height: '97',
      mass: '32',
      gender: 'n/a',
      hairColor: 'n/a',
      skinColor: 'white, red',
      eyeColor: 'red',
      birthYear: 'unknown',
    },
    {
      name: 'Biggs Darklighter',
      height: '183',
      mass: '84',
      gender: 'male',
      hairColor: 'black',
      skinColor: 'light',
      eyeColor: 'brown',
      birthYear: '24BBY',
    },
    {
      name: 'Obi-Wan Kenobi',
      height: '182',
      mass: '77',
      gender: 'male',
      hairColor: 'auburn, white',
      skinColor: 'fair',
      eyeColor: 'blue-gray',
      birthYear: '57BBY',
    },
  ],
  savedFilter: 'All',
  savedArray: [
    {
      name: 'Luke Skywalker',
      height: '172',
      mass: '77',
      gender: 'male',
      hairColor: 'blond',
      skinColor: 'fair',
      eyeColor: 'blue',
      birthYear: '19BBY',
    },
    {
      name: 'Leia Organa',
      height: '150',
      mass: '49',
      gender: 'female',
      hairColor: 'brown',
      skinColor: 'light',
      eyeColor: 'brown',
      birthYear: '19BBY',
    },
    {
      name: 'R2-D2',
      height: '96',
      mass: '32',
      gender: 'n/a',
      hairColor: 'n/a',
      skinColor: 'white, blue',
      eyeColor: 'red',
      birthYear: '33BBY',
    },
  ],
  characterSelected: {
    name: 'R2-D2',
    height: '96',
    mass: '32',
    gender: 'n/a',
    hairColor: 'n/a',
    skinColor: 'white, blue',
    eyeColor: 'red',
    birthYear: '33BBY',
  },
  searchText: '',
  error: '',
};
