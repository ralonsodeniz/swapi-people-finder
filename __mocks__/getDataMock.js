export const dataFromApiMock = {
  count: 87,
  next: 'https://swapi.co/api/people/?page=2&search=',
  previous: null,
  results: [
    {
      name: 'Luke Skywalker',
      height: '172',
      mass: '77',
      hair_color: 'blond',
      skin_color: 'fair',
      eye_color: 'blue',
      birth_year: '19BBY',
      gender: 'male',
      homeworld: 'https://swapi.co/api/planets/1/',
      films: [
        'https://swapi.co/api/films/2/',
        'https://swapi.co/api/films/6/',
        'https://swapi.co/api/films/3/',
        'https://swapi.co/api/films/1/',
        'https://swapi.co/api/films/7/',
      ],
      species: ['https://swapi.co/api/species/1/'],
      vehicles: ['https://swapi.co/api/vehicles/14/', 'https://swapi.co/api/vehicles/30/'],
      starships: ['https://swapi.co/api/starships/12/', 'https://swapi.co/api/starships/22/'],
      created: '2014-12-09T13:50:51.644000Z',
      edited: '2014-12-20T21:17:56.891000Z',
      url: 'https://swapi.co/api/people/1/',
    },
    {
      name: 'C-3PO',
      height: '167',
      mass: '75',
      hair_color: 'n/a',
      skin_color: 'gold',
      eye_color: 'yellow',
      birth_year: '112BBY',
      gender: 'n/a',
      homeworld: 'https://swapi.co/api/planets/1/',
      films: [
        'https://swapi.co/api/films/2/',
        'https://swapi.co/api/films/5/',
        'https://swapi.co/api/films/4/',
        'https://swapi.co/api/films/6/',
        'https://swapi.co/api/films/3/',
        'https://swapi.co/api/films/1/',
      ],
      species: ['https://swapi.co/api/species/2/'],
      vehicles: [],
      starships: [],
      created: '2014-12-10T15:10:51.357000Z',
      edited: '2014-12-20T21:17:50.309000Z',
      url: 'https://swapi.co/api/people/2/',
    },
    {
      name: 'R2-D2',
      height: '96',
      mass: '32',
      hair_color: 'n/a',
      skin_color: 'white, blue',
      eye_color: 'red',
      birth_year: '33BBY',
      gender: 'n/a',
      homeworld: 'https://swapi.co/api/planets/8/',
      films: [
        'https://swapi.co/api/films/2/',
        'https://swapi.co/api/films/5/',
        'https://swapi.co/api/films/4/',
        'https://swapi.co/api/films/6/',
        'https://swapi.co/api/films/3/',
        'https://swapi.co/api/films/1/',
        'https://swapi.co/api/films/7/',
      ],
      species: ['https://swapi.co/api/species/2/'],
      vehicles: [],
      starships: [],
      created: '2014-12-10T15:11:50.376000Z',
      edited: '2014-12-20T21:17:50.311000Z',
      url: 'https://swapi.co/api/people/3/',
    },
    {
      name: 'Darth Vader',
      height: '202',
      mass: '136',
      hair_color: 'none',
      skin_color: 'white',
      eye_color: 'yellow',
      birth_year: '41.9BBY',
      gender: 'male',
      homeworld: 'https://swapi.co/api/planets/1/',
      films: [
        'https://swapi.co/api/films/2/',
        'https://swapi.co/api/films/6/',
        'https://swapi.co/api/films/3/',
        'https://swapi.co/api/films/1/',
      ],
      species: ['https://swapi.co/api/species/1/'],
      vehicles: [],
      starships: ['https://swapi.co/api/starships/13/'],
      created: '2014-12-10T15:18:20.704000Z',
      edited: '2014-12-20T21:17:50.313000Z',
      url: 'https://swapi.co/api/people/4/',
    },
    {
      name: 'Leia Organa',
      height: '150',
      mass: '49',
      hair_color: 'brown',
      skin_color: 'light',
      eye_color: 'brown',
      birth_year: '19BBY',
      gender: 'female',
      homeworld: 'https://swapi.co/api/planets/2/',
      films: [
        'https://swapi.co/api/films/2/',
        'https://swapi.co/api/films/6/',
        'https://swapi.co/api/films/3/',
        'https://swapi.co/api/films/1/',
        'https://swapi.co/api/films/7/',
      ],
      species: ['https://swapi.co/api/species/1/'],
      vehicles: ['https://swapi.co/api/vehicles/30/'],
      starships: [],
      created: '2014-12-10T15:20:09.791000Z',
      edited: '2014-12-20T21:17:50.315000Z',
      url: 'https://swapi.co/api/people/5/',
    },
    {
      name: 'Owen Lars',
      height: '178',
      mass: '120',
      hair_color: 'brown, grey',
      skin_color: 'light',
      eye_color: 'blue',
      birth_year: '52BBY',
      gender: 'male',
      homeworld: 'https://swapi.co/api/planets/1/',
      films: [
        'https://swapi.co/api/films/5/',
        'https://swapi.co/api/films/6/',
        'https://swapi.co/api/films/1/',
      ],
      species: ['https://swapi.co/api/species/1/'],
      vehicles: [],
      starships: [],
      created: '2014-12-10T15:52:14.024000Z',
      edited: '2014-12-20T21:17:50.317000Z',
      url: 'https://swapi.co/api/people/6/',
    },
    {
      name: 'Beru Whitesun lars',
      height: '165',
      mass: '75',
      hair_color: 'brown',
      skin_color: 'light',
      eye_color: 'blue',
      birth_year: '47BBY',
      gender: 'female',
      homeworld: 'https://swapi.co/api/planets/1/',
      films: [
        'https://swapi.co/api/films/5/',
        'https://swapi.co/api/films/6/',
        'https://swapi.co/api/films/1/',
      ],
      species: ['https://swapi.co/api/species/1/'],
      vehicles: [],
      starships: [],
      created: '2014-12-10T15:53:41.121000Z',
      edited: '2014-12-20T21:17:50.319000Z',
      url: 'https://swapi.co/api/people/7/',
    },
    {
      name: 'R5-D4',
      height: '97',
      mass: '32',
      hair_color: 'n/a',
      skin_color: 'white, red',
      eye_color: 'red',
      birth_year: 'unknown',
      gender: 'n/a',
      homeworld: 'https://swapi.co/api/planets/1/',
      films: ['https://swapi.co/api/films/1/'],
      species: ['https://swapi.co/api/species/2/'],
      vehicles: [],
      starships: [],
      created: '2014-12-10T15:57:50.959000Z',
      edited: '2014-12-20T21:17:50.321000Z',
      url: 'https://swapi.co/api/people/8/',
    },
    {
      name: 'Biggs Darklighter',
      height: '183',
      mass: '84',
      hair_color: 'black',
      skin_color: 'light',
      eye_color: 'brown',
      birth_year: '24BBY',
      gender: 'male',
      homeworld: 'https://swapi.co/api/planets/1/',
      films: ['https://swapi.co/api/films/1/'],
      species: ['https://swapi.co/api/species/1/'],
      vehicles: [],
      starships: ['https://swapi.co/api/starships/12/'],
      created: '2014-12-10T15:59:50.509000Z',
      edited: '2014-12-20T21:17:50.323000Z',
      url: 'https://swapi.co/api/people/9/',
    },
    {
      name: 'Obi-Wan Kenobi',
      height: '182',
      mass: '77',
      hair_color: 'auburn, white',
      skin_color: 'fair',
      eye_color: 'blue-gray',
      birth_year: '57BBY',
      gender: 'male',
      homeworld: 'https://swapi.co/api/planets/20/',
      films: [
        'https://swapi.co/api/films/2/',
        'https://swapi.co/api/films/5/',
        'https://swapi.co/api/films/4/',
        'https://swapi.co/api/films/6/',
        'https://swapi.co/api/films/3/',
        'https://swapi.co/api/films/1/',
      ],
      species: ['https://swapi.co/api/species/1/'],
      vehicles: ['https://swapi.co/api/vehicles/38/'],
      starships: [
        'https://swapi.co/api/starships/48/',
        'https://swapi.co/api/starships/59/',
        'https://swapi.co/api/starships/64/',
        'https://swapi.co/api/starships/65/',
        'https://swapi.co/api/starships/74/',
      ],
      created: '2014-12-10T16:16:29.192000Z',
      edited: '2014-12-20T21:17:50.325000Z',
      url: 'https://swapi.co/api/people/10/',
    },
  ],
};

export const dataProcessedMock = {
  characterCount: 87,
  nextEndpoint: 'https://swapi.co/api/people/?page=2&search=',
  previousEndpoint: null,
  searchArray: [
    {
      name: 'Luke Skywalker',
      height: '172',
      mass: '77',
      gender: 'male',
      hairColor: 'blond',
      skinColor: 'fair',
      eyeColor: 'blue',
      birthYear: '19BBY',
    },
    {
      name: 'C-3PO',
      height: '167',
      mass: '75',
      gender: 'n/a',
      hairColor: 'n/a',
      skinColor: 'gold',
      eyeColor: 'yellow',
      birthYear: '112BBY',
    },
    {
      name: 'R2-D2',
      height: '96',
      mass: '32',
      gender: 'n/a',
      hairColor: 'n/a',
      skinColor: 'white, blue',
      eyeColor: 'red',
      birthYear: '33BBY',
    },
    {
      name: 'Darth Vader',
      height: '202',
      mass: '136',
      gender: 'male',
      hairColor: 'none',
      skinColor: 'white',
      eyeColor: 'yellow',
      birthYear: '41.9BBY',
    },
    {
      name: 'Leia Organa',
      height: '150',
      mass: '49',
      gender: 'female',
      hairColor: 'brown',
      skinColor: 'light',
      eyeColor: 'brown',
      birthYear: '19BBY',
    },
    {
      name: 'Owen Lars',
      height: '178',
      mass: '120',
      gender: 'male',
      hairColor: 'brown, grey',
      skinColor: 'light',
      eyeColor: 'blue',
      birthYear: '52BBY',
    },
    {
      name: 'Beru Whitesun lars',
      height: '165',
      mass: '75',
      gender: 'female',
      hairColor: 'brown',
      skinColor: 'light',
      eyeColor: 'blue',
      birthYear: '47BBY',
    },
    {
      name: 'R5-D4',
      height: '97',
      mass: '32',
      gender: 'n/a',
      hairColor: 'n/a',
      skinColor: 'white, red',
      eyeColor: 'red',
      birthYear: 'unknown',
    },
    {
      name: 'Biggs Darklighter',
      height: '183',
      mass: '84',
      gender: 'male',
      hairColor: 'black',
      skinColor: 'light',
      eyeColor: 'brown',
      birthYear: '24BBY',
    },
    {
      name: 'Obi-Wan Kenobi',
      height: '182',
      mass: '77',
      gender: 'male',
      hairColor: 'auburn, white',
      skinColor: 'fair',
      eyeColor: 'blue-gray',
      birthYear: '57BBY',
    },
  ],
};
