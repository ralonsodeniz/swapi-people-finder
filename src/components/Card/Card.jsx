import React from 'react';
import PropTypes from 'prop-types';

import getCharImage from '../../helpers/getCharImage';

import CustomButton from '../CustomButton/CustomButton';

import {
  CardContainer,
  CardDetailsContainer,
  CardImage,
  CardDetailsTitle,
  CardDetailsButtonsContainer,
} from './Card.styles';

const Card = ({ character }) => {
  const { name } = character;

  const imageUrl = name && getCharImage(name, '240');

  return (
    <CardContainer>
      <CardImage alt="character" src={imageUrl} />
      <CardDetailsContainer>
        <CardDetailsTitle>
          {name}
          <CardDetailsButtonsContainer>
            <CustomButton type="button" variant="default" size="small" onClick={() => {}}>
              Details
            </CustomButton>
            <CustomButton type="button" variant="remove" size="small" onClick={() => {}}>
              Remove
            </CustomButton>
          </CardDetailsButtonsContainer>
        </CardDetailsTitle>
      </CardDetailsContainer>
    </CardContainer>
  );
};

Card.propTypes = {
  character: PropTypes.shape({
    name: PropTypes.string,
    height: PropTypes.number,
    mass: PropTypes.number,
    hairColor: PropTypes.string,
    skinColor: PropTypes.string,
    eyeColor: PropTypes.string,
    birthYear: PropTypes.number,
    gender: PropTypes.string,
  }).isRequired,
};

export default Card;
