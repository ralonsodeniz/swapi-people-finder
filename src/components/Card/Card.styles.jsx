import styled, { css } from 'styled-components';

import mediaQueryHelper from '../../style/media-queries';

export const CardContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  background-color: transparent;
  width: 15rem;
  height: 20rem;
  box-shadow: 4px 4px 8px 0px rgba(0, 0, 0, 0.2);
  border-radius: 5px;

  ${mediaQueryHelper(
    'small-desktop',
    css`
      width: 8rem;
      height: 10.5rem;
    `
  )}

  ${mediaQueryHelper(
    'tab-xl',
    css`
      width: 12rem;
      height: 16rem;
    `
  )};

  ${mediaQueryHelper(
    'tab-port',
    css`
      width: 8rem;
      height: 10.5rem;
    `
  )}
`;

export const CardImage = styled.img`
  object-fit: cover;
  object-position: 50% 50%;
  height: 15rem;
  border-radius: 5px 5px 0 0;

  ${mediaQueryHelper(
    'small-desktop',
    css`
      height: 8rem;
    `
  )}

  ${mediaQueryHelper(
    'tab-xl',
    css`
      height: 12rem;
    `
  )};

  ${mediaQueryHelper(
    'tab-port',
    css`
      height: 8rem;
    `
  )};
`;

export const CardDetailsContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: ${({ theme }) => theme.titleColor};
  border-radius: 0 0 5px 5px;
  height: 5rem;
  text-align: center;
  color: ${({ theme }) => theme.primaryColor};

  ${mediaQueryHelper(
    'small-desktop',
    css`
      height: 2.5rem;
    `
  )}

  ${mediaQueryHelper(
    'tab-xl',
    css`
      height: 4rem;
    `
  )};

  ${mediaQueryHelper(
    'tab-port',
    css`
      height: 2.5rem;
    `
  )};
`;

export const CardDetailsTitle = styled.h3`
  font-weight: bold;
  font-size: ${({ theme }) => theme.midFontSize};

  ${mediaQueryHelper(
    'small-desktop',
    css`
      font-size: ${({ theme }) => theme.smallFontSize};
    `
  )}

  ${mediaQueryHelper(
    'tab-xl',
    css`
      font-size: ${({ theme }) => theme.defaultFontSize};
    `
  )};

  ${mediaQueryHelper(
    'tab-port',
    css`
      font-size: ${({ theme }) => theme.smallFontSize};
    `
  )};
`;

export const CardDetailsButtonsContainer = styled.div`
  display: flex;
  justify-content: space-evenly;
`;
