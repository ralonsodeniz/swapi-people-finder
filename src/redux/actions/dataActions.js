/* eslint-disable no-console */
import DATA from '../types/dataTypes';
import { getPageDataFromAPI } from '../helpers/dataHelpers';

export const toggleDataLoading = loadingState => ({
  type: DATA.TOGGLE_DATA_LOADING,
  payload: loadingState,
});

export const fetchApiSuccess = searchData => ({
  type: DATA.FETCH_API_SUCCESS,
  payload: searchData,
});

export const fetchApiFailure = error => ({
  type: DATA.FETCH_API_FAILURE,
  payload: error,
});

export const saveCharacter = character => ({
  type: DATA.SAVE_CHARACTER,
  payload: character,
});

export const removeCharacter = characterName => ({
  type: DATA.REMOVE_CHARACTER,
  payload: characterName,
});

export const selectCharacter = character => ({
  type: DATA.SELECT_CHARACTER,
  payload: character,
});

export const setFilter = filter => ({
  type: DATA.SET_FILTER,
  payload: filter,
});

export const setSeachText = searchText => ({
  type: DATA.SET_SEARCH_TEXT,
  payload: searchText,
});

export const fetchApiStart = url => async dispatch => {
  dispatch(toggleDataLoading(true));
  try {
    const searchData = await getPageDataFromAPI(url);
    dispatch(fetchApiSuccess(searchData));
  } catch (error) {
    dispatch(fetchApiFailure(error.message));
    console.log(error);
  }
};
