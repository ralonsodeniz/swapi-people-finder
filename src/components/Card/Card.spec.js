import React from 'react';
import { shallow } from 'enzyme';
import * as Redux from 'react-redux';
import configureStore from 'redux-mock-store';

import { c3p0, darthVader } from '../../../__mocks__/charInfoMock';
import { selectCharacter, removeCharacter } from '../../redux/actions/dataActions';

import Card from './Card';
import { CardImage, CardDetailsTitle } from './Card.styles';

describe('<Card /> tests', () => {
  let wrapper;
  let store;
  const getWrapper = mockProps => shallow(<Card character={mockProps} store={store} />);
  const selectCharacterAction = selectCharacter(c3p0);
  const removeCharacterAction = removeCharacter(c3p0.name);

  beforeEach(() => {
    store = configureStore()({});
    jest.spyOn(Redux, 'useDispatch').mockImplementation(() => store.dispatch);
    wrapper = getWrapper(c3p0);
    store.clearActions();
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has one img with src to C-3PO portrait', () => {
    const img = wrapper.find(CardImage);
    expect(img.length).toEqual(1);
    const imageUrl = img.prop('src');
    expect(imageUrl).toEqual('http://facetheforce.today/c3po/240');
  });

  it('has one img with src to Darth Vader portrait', () => {
    wrapper = getWrapper(darthVader);
    const img = wrapper.find(CardImage);
    expect(img.length).toEqual(1);
    const imageUrl = img.prop('src');
    expect(imageUrl).toEqual('http://facetheforce.today/darthvader/240');
  });

  it('has one title with text C-3PO', () => {
    const title = wrapper.find(CardDetailsTitle);
    expect(title.length).toEqual(1);
    const titleText = title.text();
    expect(titleText.slice(0, titleText.search('<'))).toEqual('C-3PO');
  });

  it('has two buttons', () => {
    const buttons = wrapper.find('CustomButton');
    expect(buttons.length).toEqual(2);
  });

  it('has one default small size Detail button that triggrs selectCharacter action', () => {
    const button = wrapper.find('CustomButton').at(0);
    const buttonText = button.shallow().text();
    expect(buttonText).toEqual('Details');
    const buttonSize = button.prop('size');
    expect(buttonSize).toEqual('small');
    const buttonVariant = button.prop('variant');
    expect(buttonVariant).toEqual('default');
    button.simulate('click');
    const actions = store.getActions();
    expect(actions).toEqual([selectCharacterAction]);
  });

  it('has one remove small size Remove button that triggrs removeCharacterAction action', () => {
    const button = wrapper.find('CustomButton').at(1);
    const buttonText = button.shallow().text();
    expect(buttonText).toEqual('Remove');
    const buttonSize = button.prop('size');
    expect(buttonSize).toEqual('small');
    const buttonVariant = button.prop('variant');
    expect(buttonVariant).toEqual('remove');
    button.simulate('click');
    const actions = store.getActions();
    expect(actions).toEqual([removeCharacterAction]);
  });
});
