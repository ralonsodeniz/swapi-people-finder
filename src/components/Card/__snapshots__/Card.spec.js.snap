// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> tests matches snapshot 1`] = `
<styled.div>
  <styled.img
    alt="character"
    src="http://facetheforce.today/c3po/240"
  />
  <styled.div>
    <styled.h3>
      C-3PO
      <styled.div>
        <CustomButton
          collapse={false}
          onClick={[Function]}
          selected={false}
          size="small"
          type="button"
          variant="default"
        >
          Details
        </CustomButton>
        <CustomButton
          collapse={false}
          onClick={[Function]}
          selected={false}
          size="small"
          type="button"
          variant="remove"
        >
          Remove
        </CustomButton>
      </styled.div>
    </styled.h3>
  </styled.div>
</styled.div>
`;
